### PR TITLE
AMDGPU: Add test for 16 bit unsigned scratch offsets

### PR DESCRIPTION
**Pull request by petar-avramovic**
Large scratch offset with one on highest bit selected as negative,
negative offset has same binary representation in 16 bits as large
unsigned offset.

### CODE REVIEW

**Comment by petar-avramovic** [comment]
* **#110256** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/110256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#110255** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/110255?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @petar-avramovic and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by jayfoad — APPROVED** [review]

File: llvm/test/CodeGen/AMDGPU/GlobalISel/flat-scratch.ll [hunk 1]
```diff
@@ -1510,4 +1510,243 @@ bb:
   ret void
 }
 
+define amdgpu_gs void @sgpr_base_large_offset(ptr addrspace(1) %out, ptr addrspace(5) inreg %sgpr_base) {
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/flat-scratch.ll:1513]: We should probably remove this file, and add extra run lines to the other file instead. But that would be a separate patch.

**Comment by petar-avramovic** [comment]
### Merge activity

* **Sep 30, 4:37 AM EDT**: @petar-avramovic started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/110255).
